crt_portal/cts_forms/views.py

```diff
@@ -988,41 +994,58 @@ def get(self, request, id=None):
         selected_all = request.GET.get('all', '') == 'all'
         if selected_all:
             requested_query = reconstruct_query(query_string)
+            selected_report_args = 'all=all'
         else:
             requested_query = Report.objects.filter(pk__in=ids)
+            selected_report_args = self.reconstruct_id_args(ids)
 
-        data = self.get_report_data(requested_query)
+        if requested_query.count() > 500:
+            raise PermissionDenied
```

Comment:
is PermissionDenied right for this? maybe this is a BadRequest?

Reply:
I used PermissionDenied because that's what we use for bulk actions currently when the query count is too high. Looking into it it does seem like BadRequest is more appropriate. Wdyt about changing the exception for bulk actions as well?

Reply:
I think that makes sense. I don't see anywhere in the frontend bulk actions that PermissionDenied is referenced (403, etc)